controllers/gitrepository_controller.go

```diff
@@ -786,6 +792,8 @@ func (r *GitRepositoryReconciler) gitCheckout(ctx context.Context,
 		conditions.MarkTrue(obj, sourcev1.FetchFailedCondition, e.Reason, e.Err.Error())
 		return nil, e
 	}
+	gitReader.Close()
```

Comment:
We can defer this call after the client creation error check, this would prevent a memory leak being introduced in case `NewClient` ever needs to create non garbage collectable objects.